src/options/views/tab-installed.vue

```diff
@@ -378,6 +380,39 @@ export default {
         if (step) await new Promise(resolve => setTimeout(resolve));
       }
     },
+    performSearch(scripts) {
+      let searchRE;
+      let searchError;
+      const [, expr, flags] = this.search.match(/^\/(.+?)\/(\w*)$|$/);
+      if (expr) {
+        try {
+          searchRE = new RegExp(expr, `i${flags.replace('i', '')}`);
+        } catch (err) {
+          searchError = err.message;
+        }
+      }
+      this.searchError = searchError || null;
+      const lowerSearch = !searchRE && (this.search || '').toLowerCase();
+      scripts.forEach(({ $cache }) => {
+        $cache.show = $cache.search.some((item, i, arr) => {
+          if (typeof item !== 'string') {
+            item = item[0].toLowerCase();
+            arr[i] = item;
+          }
+          return searchRE ? searchRE.test(item) : item.includes(lowerSearch);
+        });
+      });
+    },
+    async scheduleSearch() {
+      if (!scriptCodeRetrieved) {
+        scriptCodeRetrieved = true;
+        const { scripts } = this.store;
+        const ids = scripts.map(script => `code:${script.props.id}`);
+        const data = await browser.storage.local.get(ids);
```

Comment:
I think `browser.storage.local` should not be used directly outside background, because this requires the options page to know about the data structure defined in background, maybe it's better to load from background with a command.

Reply:
I thought about that too, but I **really** want to avoid the triple processing of a huge chunk of data: local.get() parses into JS object, messaging stringifies it, the receiver parses it again.

What if I put those storage prefixes in a separate file like `storage-consts.js`?

Reply:
In my opinion, the options page should not know about the implementation details of storage. This makes the application more portable and easier to port to other platforms.

And I don't think it's a big deal to transport some large strings during a search by user, especially when it's already used in the script editor and installation confirming.

We can even pass the search text to background and return the filtered items asynchronously if you really care about the large messages.

Reply:
>more portable
It shouldn't be an issue because a compatibility polyfill for browser.storage would process the key names regardless of where the request is performed.

>And I don't think it's a big deal to transport some large strings during a search by user, especially when it's already used in the script editor and installation confirming.

Why? I have 500kB in my 100 scripts. My scripts are rather small, but some people may have much bigger scripts. The script editor/confirmer only uses one script, not all of them.

>We can even pass the search text to background
That would be asynchronous so it'll add the unnecessary complexity to the options page. And the background page will have to keep those things in a cache => more complexity.

Reply:
> That would be asynchronous so it'll add the unnecessary complexity to the options page. And the background page will have to keep those things in a cache => more complexity.

The scripts are already kept in memory because they are possibly needed whenever a page is loaded. So I think the *complexity* is actually just a `filter` in the background.

> It shouldn't be an issue because a compatibility polyfill for browser.storage would process the key names regardless of where the request is performed.

It is an issue because `browser.storage` or its alternative is not always accessible from a place other than background. For example, Maxthon again.

Maybe it's a good idea to have a separate `common/scripts.js` to load and dump scripts. We can check environment in it and fetch scripts in a proper way.

ps. Maybe we should drop support for Maxthon in the future and just follow the WebExtension standard. And there are some more, Edge, Safari, etc.

Reply:
> The scripts are already kept in memory

Only the script info object AFAICT, not the `code`.

> browser.storage or its alternative is not always accessible 

No problem: the polyfill should simply imitate browser.storage.

> separate common/scripts.js to load and dump scripts.

Maybe, but the only part that the options page needs is the code, not the rest of the info which is cached and processed inside the background page.

Reply:
I prefer using an API from options page, rather than the level `browser.storage`, even with prefix constants. How about moving some of the methods in `src/background/utils/db` to common?

Reply:
I'll try, okay.